app_spec/zomes/simple/code/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+#![feature(try_from)]
+#![warn(unused_extern_crates)]
+#[macro_use]
+extern crate hdk;
+#[macro_use]
+extern crate serde_derive;
+#[macro_use]
+extern crate holochain_core_types_derive;
+
+use hdk::{
+    entry_definition::ValidatingEntryType,
+    error::ZomeApiResult,
+};
+use hdk::holochain_core_types::{
+    cas::content::Address,
+    dna::entry_types::Sharing,
+    error::HolochainError,
+    json::JsonString,
+    entry::Entry
+};
+
+use hdk::holochain_wasm_utils::api_serialization::get_links::GetLinksResult;
+
+
+// see https://developer.holochain.org/api/0.0.18-alpha1/hdk/ for info on using the hdk library
+
+// This is a sample zome that defines an entry type "MyEntry" that can be committed to the
+// agent's chain via the exposed function create_my_entry
+
+#[derive(Serialize, Deserialize, Debug, DefaultJson,Clone)]
+pub struct Simple {
+    content: String,
+}
+
+impl Simple 
+{
+    pub fn new(content:String) -> Simple
+    {
+        Simple{content}
+    }
+}
+
+fn simple_entry(content: String) -> Entry {
+    Entry::App("simple".into(), Simple::new(content).into())
+}
+
+
+pub fn handle_create_my_link(base: Address,content : String) -> ZomeApiResult<()> {
+    let address = hdk::commit_entry(&simple_entry(content))?;
```

Comment:
We need to commit an address before we can link it to the link_entry, I wouldn't want to seperate the two because it would cause a lot of noise for the app_spec part since we care mostly about the links rather than the entry